### PR TITLE
feat(Avatar)!: convert avatar sizes to using numbers over string enum

### PR DESCRIPTION
**Pull request by MattSchiller**
### Overview

Changed the <Avatar /> component to using numbers directly over the string enum to give greater flexibility with size extensions

### PR Checklist

- [X] Related to designs: https://www.figma.com/file/SwfwQZmV4jtK68QaCa89h3/CC-Docs-v2---User-Recognition?node-id=452%3A61680
- [X] Related to JIRA ticket: REACH-1264
- [ ] I have run this code to verify it works
- [X] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.5.76-alpha.241a27.0<br/>@codecademy/gamut-labs@19.26.5-alpha.241a27.0<br/>@codecademy/styleguide@44.5.1-alpha.241a27.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6140c7d29da7a000ad2b9066--gamut-preview.netlify.app>https://6140c7d29da7a000ad2b9066--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6140c7d29da7a000ad2b9066>Deploy Logs</a></p>

File: packages/gamut-labs/src/Avatar/index.tsx [hunk 1]
```diff
@@ -99,10 +94,6 @@ export const Avatar: React.FC<AvatarProps> = ({
     disableDropshadow={disableDropshadow}
     data-testid="avatar-container"
   >
-    <Image
-      width={avatarSizes[size]}
-      height={avatarSizes[size]}
-      {...avatarImageProps}
-    />
+    <Image width={`${size}px`} height={`${size}px`} {...avatarImageProps} />
```
Comment (codecaaron) [inline packages/gamut-labs/src/Avatar/index.tsx:97]: The shorthand for both `width` and `height` together is annoyingly `dimensions` because of other prop conflicts for buttons etc but it also will do all of the transformations for you so we can just pass numbers here.

```suggestion
    <Image dimensions={size} {...avatarImageProps} />
```

File: packages/gamut-labs/src/Avatar/index.tsx [hunk 2]
```diff
@@ -62,9 +62,9 @@ export type AvatarBaseProps = {
   disableDropshadow?: boolean;
 
   /**
-   * Size of the Avatar; small = 32x32, medium = 118x118
+   * Size of the Avatar.
    */
-  size?: 'small' | 'medium';
+  size?: 32 | 48 | 64 | 118;
```
Comment (codecaaron) [inline packages/gamut-labs/src/Avatar/index.tsx:67]: Let's add system props to img here so we can also do some media query syntax.

```tsx
import { ResponsiveProp } from '@codecademy/variance';
import { system, styledOptions } from '@codecademy/gamut-styles';

const Image = styled('img', styledOptions<'img'>())(system.layout);
```

```suggestion
  size?: ResponsiveProp<32 | 48 | 64 | 118>;
```

**Review by codecaaron — COMMENTED** [review]
Added some low lift suggestions - these will break your tests as the properties won't be forwarded anymore with this change.

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.5.76-alpha.a703a9.0<br/>@codecademy/gamut-labs@19.26.5-alpha.a703a9.0<br/>@codecademy/styleguide@44.5.1-alpha.a703a9.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6140d46ac9cc3300d31d8a88--gamut-preview.netlify.app>https://6140d46ac9cc3300d31d8a88--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6140d46ac9cc3300d31d8a88>Deploy Logs</a></p>

File: packages/gamut-labs/src/Avatar/index.tsx [hunk 3]
```diff
@@ -1,10 +1,11 @@
 import { VisualTheme } from '@codecademy/gamut';
-import { theme } from '@codecademy/gamut-styles';
+import { styledOptions, system, theme } from '@codecademy/gamut-styles';
+import { ResponsiveProp } from '@codecademy/variance';
 import { css } from '@emotion/react';
 import styled from '@emotion/styled';
 import React from 'react';
 
-const Image = styled.img();
+const Image = styled('img', styledOptions<'img'>())(system.layout);
```
Comment (christian-dinh) [inline packages/gamut-labs/src/Avatar/index.tsx:8]: what exactly is this doing? is this related to `ResponsiveProp`?

**Review by christian-dinh — COMMENTED** [review]
lgtm, I'll defer to web plat for official approval tho!

File: packages/gamut-labs/src/Avatar/index.tsx (hunk 3 above)
Comment (MattSchiller) [reply packages/gamut-labs/src/Avatar/index.tsx:8]: Yeap, it seems to allow the `dimensions` prop to be correctly typed for responsive object formatting.

File: packages/gamut-labs/src/Avatar/__tests__/avatar-test.tsx [hunk 4]
```diff
@@ -21,20 +21,12 @@ describe('Avatar', () => {
...
 
   it('when a "size" prop is passed, the <img/> height and width attributes are set accordingly', () => {
-    const wrapper = mount(<Avatar src="" alt="" size="small" />);
-    expect(
-      wrapper.find(
-        `img[height="${avatarSizes.small}"][width="${avatarSizes.small}"]`
-      )
-    ).toHaveLength(1);
+    const wrapper = mount(<Avatar src="" alt="" size={32} />);
+    expect(wrapper.find('Image').props()).toMatchObject({ dimensions: 32 });
```
Comment (codecaaron) [inline packages/gamut-labs/src/Avatar/__tests__/avatar-test.tsx:25]: NITTIEST ⛳ 
```suggestion
    expect(wrapper.find('Image').prop('dimensions')).toBe(32);
```

**Review by codecaaron — APPROVED** [review]
🥂 - LGTM

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.5.76-alpha.a59461.0<br/>@codecademy/gamut-labs@19.26.5-alpha.a59461.0<br/>@codecademy/styleguide@44.5.1-alpha.a59461.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61423003fa933f1152cfef1b--gamut-preview.netlify.app>https://61423003fa933f1152cfef1b--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61423003fa933f1152cfef1b>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.5.79-alpha.ea3e85.0<br/>@codecademy/gamut-labs@19.26.8-alpha.ea3e85.0<br/>@codecademy/styleguide@44.6.1-alpha.ea3e85.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61424e2564f8fd32185527ea--gamut-preview.netlify.app>https://61424e2564f8fd32185527ea--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61424e2564f8fd32185527ea>Deploy Logs</a></p>
